source/dotnet/Test/AdaptiveCards.Test/AdaptiveCardApiTests.cs

```diff
@@ -94,7 +94,18 @@ public void TestParsingInvalidCardMissingVersion()
     }
   ]
 }";
+            // By default we should be throwing an exception when no version present.
             Assert.ThrowsException<AdaptiveSerializationException>(() => AdaptiveCard.FromJson(json));
+
+            // Or we can supply a replacement version to use
+            AdaptiveCard.OnDeserializingMissingVersion = () => new AdaptiveSchemaVersion(0, 5);
+            var card = AdaptiveCard.FromJson(json);
+            Assert.AreEqual(new AdaptiveSchemaVersion(0, 5), card.Card.Version);
+
+            // But make sure that if the callback throws an exception that it is allowed through
+            AdaptiveCard.OnDeserializingMissingVersion = () => throw new Exception("test exception");
```

Comment:
>st exception" [](start = 86, length = 13)

nit: this could be better :)

Reply:
this is just a unit test :)

---
In reply to: [325817981](https://github.com/microsoft/AdaptiveCards/pull/3462#discussion_r325817981) [](ancestors = 325817981)

Reply:
We should treat unit test code no different than production code! 

---
In reply to: [325824909](https://github.com/microsoft/AdaptiveCards/pull/3462#discussion_r325824909) [](ancestors = 325824909,325817981)